### PR TITLE
Bec214 airkeeper unit tests

### PR DESCRIPTION
**Pull request by acenolaza**
I've added unit tests for all functions in the evm directory. I thought about adding tests to the handlers but then I realized that these are going to change a lot by the refactors planned for each one. ([rrp refactor](https://api3dao.atlassian.net/browse/BEC-238) and [psp with lambda invocation](https://api3dao.atlassian.net/browse/BEC-213)).

### CODE REVIEW

File: src/evm/initialize-provider.test.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+      AirnodeRrp: '0x5FbDB2315678afecb367f032d93F642f64180aa3',
+      RrpBeaconServer: '0xB7f8BC63BbcaD18155201308C8f3540b07f84F5e',
+      DapiServer: '0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0',
+    },
+    id: '31337',
+    providers: { local: { url: 'http://127.0.0.1:8545' } },
+    type: 'evm',
+    options: {
+      txType: 'eip1559',
+      baseFeeMultiplier: '2',
```
Comment (vponline) [inline src/evm/initialize-provider.test.ts:22]: The gas price types will change to numbers in airnode v0.5, but these will be updated in the refactoring PRs.

**Review by vponline — APPROVED** [review]
I think these look good to go :+1:  (minor changes coming with the refactoring PRs)

File: src/evm/initialize-provider.test.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+      priorityFee: { value: '3.12', unit: 'gwei' },
+    },
+  };
+  const providerUrl = 'http://localhost:8545';
```
Comment (amarthadan) [inline src/evm/initialize-provider.test.ts:26]: Can you use the same URL?

**Review by amarthadan — CHANGES_REQUESTED** [review]
Nice work :+1:

File: src/evm/check-conditions.test.ts [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+      },
+    ]);
+    expect(data).toEqual(false);
+  });
+
+  it('returns false with error log if conditions result is null', async () => {
+    const conditionPspBeaconUpdateNullSpy = jest
+      .fn()
+      .mockImplementation((_subscriptionId: string, _encodedFulfillmentData: string, _conditionParameters: string) =>
+        Promise.resolve(null)
```
[the remark this answers is not in the document]
Comment (acenolaza) [reply src/evm/check-conditions.test.ts:172]: I'm just trying to cover these lines: https://github.com/api3dao/airkeeper/blob/bec214-airkeeper-unit-tests/src/evm/check-conditions.ts#L49-L53

Maybe `undefined` might be a more accurate return value 🤔

[the remark this answers is not in the document]
Comment (amarthadan) [reply src/evm/check-conditions.test.ts:172]: I think you won't need to worry about that once you rebase :smile: That's why I didn't see what you're trying to test, I was looking at the current main :smile:

[the remark this answers is not in the document]
Comment (acenolaza) [reply src/evm/check-conditions.test.ts]: LOL yes, after rebase these are the lines https://github.com/api3dao/airkeeper/blob/bec214-airkeeper-unit-tests/src/evm/check-conditions.ts#L53-L57 and I think the test might still apply

**Review by amarthadan — APPROVED** [review]

**Pull request merged**